configs/recommended.js

```diff
@@ -0,0 +1,14 @@
+"use strict"
+
+const mod = require("../lib/index.js")
+const {globals, rules, overrides} = mod.configs["recommended"]
+
+module.exports = {
+    plugins: { n: mod },
+    languageOptions: {
+        sourceType: "module",
+        globals,
+    },
+    rules,
+    overrides,
```

Comment:
but the flat config does not have `overrides`.

Reply:
Can we use `compat` here? Or write overrides in a different way? As an array items, like FlatESlint suggests, for example.

The thing is auto detecting of module type this what blocking me from upgrade to FlatESlint, since I have both types supported and think that one place to set module type in `package.json` is enough (like in legacy ESLint config plugin version)

Reply:
well, I think we can add 2 more presets:

* recommended: detecting the "type" field in `package.json`, use `recommended-script` if it's `commonjs/undefined`; use `recommended-module` if it's `module`.
* mixed-esm-and-cjs(a better name? 😏): exporting a flat configs like the eslintrc-style config array.

Reply:
Yes, it would be awesome!